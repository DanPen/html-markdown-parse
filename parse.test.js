const parse = require('./parse')

// Test that the node iterator works as expected (on whiteboard).
// Test mardown

test('tree parse b->t', () => {
    const tree = parse.toTree(`<b>Hello World!</b>`)
    expect(tree).toMatchSnapshot()
})

test('tree parse b->t | t', () => {
    const tree = parse.toTree(`<b>Hello World!</b> Test`)
    expect(tree).toMatchSnapshot()
})

test('tree parse b->i->t', () => {
    const tree = parse.toTree(`<b><i>Hello World!</i></b>`)
    expect(tree).toMatchSnapshot()
})

test('tree parse a->t', () => {
    const tree = parse.toTree(`<a href="https://google.com">Google</a>`)
    expect(tree).toMatchSnapshot()
})

test('tree parse b->t | br | br | i->t', () => {
    const tree = parse.toTree(`<b>hello</b><br><br> <i>world</i>`)
    expect(tree).toMatchSnapshot()
})

test('tree iterator', () => {
    const tree = parse.toTree(`<i><b><i>Hello World!</i> Here is my <a>Link</a>1</b>2</i>3`)
    expect([...tree]).toMatchSnapshot()
    // for (node of tree) {
    //     if (node instanceof parse.Node) {
    //         console.log(node.tag)
    //     }
    //     else {
    //         console.log(node)
    //     }
    // }
})

test('markdown b->t|i->t|a->t', () => {
    const markdown = parse.toMarkdown(`<b>foo <i>yo <a href="https://google.com">Google</a> </i></b>`)
    expect(markdown).toMatchSnapshot()
})
test('markdown b->t | br | br | i->t', () => {
    const markdown = parse.toMarkdown(`<b>hello</b><br><br> <i>world</i>`)
    expect(markdown).toMatchSnapshot()
})
test('markdown b|a', () => {
    const markdown = parse.toMarkdown(`<b>hello</b> <a href='https://bj.cards'>world</a>`)
    expect(markdown).toMatchSnapshot()
})