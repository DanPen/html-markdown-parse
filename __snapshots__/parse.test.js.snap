// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`markdown b|a 1`] = `"**hello** [world](https://bj.cards)"`;

exports[`markdown b->t | br | br | i->t 1`] = `"**hello** *world*"`;

exports[`markdown b->t|i->t|a->t 1`] = `"**foo *yo [Google](https://google.com) ***"`;

exports[`tree iterator 1`] = `
Array [
  Object {
    "event": "start",
    "node": Node {
      "attributes": undefined,
      "children": Array [
        Node {
          "attributes": undefined,
          "children": Array [
            Node {
              "attributes": undefined,
              "children": Array [
                "Hello World!",
              ],
              "parent": [Circular],
              "tag": "i",
            },
            " Here is my ",
            Node {
              "attributes": undefined,
              "children": Array [
                "Link",
              ],
              "parent": [Circular],
              "tag": "a",
            },
            "1",
          ],
          "parent": [Circular],
          "tag": "b",
        },
        "2",
      ],
      "parent": Node {
        "attributes": undefined,
        "children": Array [
          [Circular],
          "3",
        ],
        "parent": undefined,
        "tag": undefined,
      },
      "tag": "i",
    },
  },
  Object {
    "event": "start",
    "node": Node {
      "attributes": undefined,
      "children": Array [
        Node {
          "attributes": undefined,
          "children": Array [
            "Hello World!",
          ],
          "parent": [Circular],
          "tag": "i",
        },
        " Here is my ",
        Node {
          "attributes": undefined,
          "children": Array [
            "Link",
          ],
          "parent": [Circular],
          "tag": "a",
        },
        "1",
      ],
      "parent": Node {
        "attributes": undefined,
        "children": Array [
          [Circular],
          "2",
        ],
        "parent": Node {
          "attributes": undefined,
          "children": Array [
            [Circular],
            "3",
          ],
          "parent": undefined,
          "tag": undefined,
        },
        "tag": "i",
      },
      "tag": "b",
    },
  },
  Object {
    "event": "start",
    "node": Node {
      "attributes": undefined,
      "children": Array [
        "Hello World!",
      ],
      "parent": Node {
        "attributes": undefined,
        "children": Array [
          [Circular],
          " Here is my ",
          Node {
            "attributes": undefined,
            "children": Array [
              "Link",
            ],
            "parent": [Circular],
            "tag": "a",
          },
          "1",
        ],
        "parent": Node {
          "attributes": undefined,
          "children": Array [
            [Circular],
            "2",
          ],
          "parent": Node {
            "attributes": undefined,
            "children": Array [
              [Circular],
              "3",
            ],
            "parent": undefined,
            "tag": undefined,
          },
          "tag": "i",
        },
        "tag": "b",
      },
      "tag": "i",
    },
  },
  Object {
    "event": "leaf",
    "node": "Hello World!",
  },
  Object {
    "event": "end",
    "node": Node {
      "attributes": undefined,
      "children": Array [
        "Hello World!",
      ],
      "parent": Node {
        "attributes": undefined,
        "children": Array [
          [Circular],
          " Here is my ",
          Node {
            "attributes": undefined,
            "children": Array [
              "Link",
            ],
            "parent": [Circular],
            "tag": "a",
          },
          "1",
        ],
        "parent": Node {
          "attributes": undefined,
          "children": Array [
            [Circular],
            "2",
          ],
          "parent": Node {
            "attributes": undefined,
            "children": Array [
              [Circular],
              "3",
            ],
            "parent": undefined,
            "tag": undefined,
          },
          "tag": "i",
        },
        "tag": "b",
      },
      "tag": "i",
    },
  },
  Object {
    "event": "leaf",
    "node": " Here is my ",
  },
  Object {
    "event": "start",
    "node": Node {
      "attributes": undefined,
      "children": Array [
        "Link",
      ],
      "parent": Node {
        "attributes": undefined,
        "children": Array [
          Node {
            "attributes": undefined,
            "children": Array [
              "Hello World!",
            ],
            "parent": [Circular],
            "tag": "i",
          },
          " Here is my ",
          [Circular],
          "1",
        ],
        "parent": Node {
          "attributes": undefined,
          "children": Array [
            [Circular],
            "2",
          ],
          "parent": Node {
            "attributes": undefined,
            "children": Array [
              [Circular],
              "3",
            ],
            "parent": undefined,
            "tag": undefined,
          },
          "tag": "i",
        },
        "tag": "b",
      },
      "tag": "a",
    },
  },
  Object {
    "event": "leaf",
    "node": "Link",
  },
  Object {
    "event": "end",
    "node": Node {
      "attributes": undefined,
      "children": Array [
        "Link",
      ],
      "parent": Node {
        "attributes": undefined,
        "children": Array [
          Node {
            "attributes": undefined,
            "children": Array [
              "Hello World!",
            ],
            "parent": [Circular],
            "tag": "i",
          },
          " Here is my ",
          [Circular],
          "1",
        ],
        "parent": Node {
          "attributes": undefined,
          "children": Array [
            [Circular],
            "2",
          ],
          "parent": Node {
            "attributes": undefined,
            "children": Array [
              [Circular],
              "3",
            ],
            "parent": undefined,
            "tag": undefined,
          },
          "tag": "i",
        },
        "tag": "b",
      },
      "tag": "a",
    },
  },
  Object {
    "event": "leaf",
    "node": "1",
  },
  Object {
    "event": "end",
    "node": Node {
      "attributes": undefined,
      "children": Array [
        Node {
          "attributes": undefined,
          "children": Array [
            "Hello World!",
          ],
          "parent": [Circular],
          "tag": "i",
        },
        " Here is my ",
        Node {
          "attributes": undefined,
          "children": Array [
            "Link",
          ],
          "parent": [Circular],
          "tag": "a",
        },
        "1",
      ],
      "parent": Node {
        "attributes": undefined,
        "children": Array [
          [Circular],
          "2",
        ],
        "parent": Node {
          "attributes": undefined,
          "children": Array [
            [Circular],
            "3",
          ],
          "parent": undefined,
          "tag": undefined,
        },
        "tag": "i",
      },
      "tag": "b",
    },
  },
  Object {
    "event": "leaf",
    "node": "2",
  },
  Object {
    "event": "end",
    "node": Node {
      "attributes": undefined,
      "children": Array [
        Node {
          "attributes": undefined,
          "children": Array [
            Node {
              "attributes": undefined,
              "children": Array [
                "Hello World!",
              ],
              "parent": [Circular],
              "tag": "i",
            },
            " Here is my ",
            Node {
              "attributes": undefined,
              "children": Array [
                "Link",
              ],
              "parent": [Circular],
              "tag": "a",
            },
            "1",
          ],
          "parent": [Circular],
          "tag": "b",
        },
        "2",
      ],
      "parent": Node {
        "attributes": undefined,
        "children": Array [
          [Circular],
          "3",
        ],
        "parent": undefined,
        "tag": undefined,
      },
      "tag": "i",
    },
  },
  Object {
    "event": "leaf",
    "node": "3",
  },
]
`;

exports[`tree parse a->t 1`] = `
Node {
  "attributes": undefined,
  "children": Array [
    Node {
      "attributes": Object {
        "href": "https://google.com",
      },
      "children": Array [
        "Google",
      ],
      "parent": [Circular],
      "tag": "a",
    },
  ],
  "parent": undefined,
  "tag": undefined,
}
`;

exports[`tree parse b->i->t 1`] = `
Node {
  "attributes": undefined,
  "children": Array [
    Node {
      "attributes": undefined,
      "children": Array [
        Node {
          "attributes": undefined,
          "children": Array [
            "Hello World!",
          ],
          "parent": [Circular],
          "tag": "i",
        },
      ],
      "parent": [Circular],
      "tag": "b",
    },
  ],
  "parent": undefined,
  "tag": undefined,
}
`;

exports[`tree parse b->t | br | br | i->t 1`] = `
Node {
  "attributes": undefined,
  "children": Array [
    Node {
      "attributes": undefined,
      "children": Array [
        "hello",
      ],
      "parent": [Circular],
      "tag": "b",
    },
    Node {
      "attributes": undefined,
      "children": Array [],
      "parent": [Circular],
      "tag": "br",
    },
    Node {
      "attributes": undefined,
      "children": Array [],
      "parent": [Circular],
      "tag": "br",
    },
    " ",
    Node {
      "attributes": undefined,
      "children": Array [
        "world",
      ],
      "parent": [Circular],
      "tag": "i",
    },
  ],
  "parent": undefined,
  "tag": undefined,
}
`;

exports[`tree parse b->t | t 1`] = `
Node {
  "attributes": undefined,
  "children": Array [
    Node {
      "attributes": undefined,
      "children": Array [
        "Hello World!",
      ],
      "parent": [Circular],
      "tag": "b",
    },
    " Test",
  ],
  "parent": undefined,
  "tag": undefined,
}
`;

exports[`tree parse b->t 1`] = `
Node {
  "attributes": undefined,
  "children": Array [
    Node {
      "attributes": undefined,
      "children": Array [
        "Hello World!",
      ],
      "parent": [Circular],
      "tag": "b",
    },
  ],
  "parent": undefined,
  "tag": undefined,
}
`;
